closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -72,6 +76,33 @@
         }
     }
 
+    private static final class CBCCleanerRunnable implements Runnable {
+        private final int ctxIndx;
+        private final long nativeContext;
+
+        public CBCCleanerRunnable(int ctxIndx, long nativeContext) {
+            this.ctxIndx = ctxIndx;
+            this.nativeContext = nativeContext;
+        }
+
+        @Override
+        public void run() {
+            /*
+             * Release the CBC context.
+             */
+            if (ctxIndx == -1) {
+                long ret = nativeCrypto.DestroyContext(nativeContext);
```

Comment:
Previously, this call occurred within a `synchronized (NativeCipherBlockChaining.class)` block: Do we *know* that it is safe for multiple threads to call `nativeCrypto.DestroyContext()` concurrently?

Reply:
Thanks, i will change to both `avStack.push(ctxIndx)` and `nativeCrypto.DestroyContext(nativeContext)` within a synchronized (NativeCipherBlockChaining.class) for safety.